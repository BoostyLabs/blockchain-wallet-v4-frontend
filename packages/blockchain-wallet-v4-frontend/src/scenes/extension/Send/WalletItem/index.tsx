import React from 'react'
import { IconBlockchainCircle } from '@blockchain-com/icons'
import styled from 'styled-components'

import { Text } from 'blockchain-info-components'

const WalletWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  color: white;
  cursor: pointer;
`

const AccountInfo = styled.div`
  margin-left: 16px;
  display: flex;
  flex-direction: column;
`

const WalletItem: React.FC<{ walletAddress: string }> = ({ walletAddress }) => {
  return (
    <WalletWrapper id='walletItem'>
      <IconBlockchainCircle height='32px' color='#473BCB' width='32px' />
      <AccountInfo>
        <Text size='12px' lineHeight='18px' style={{ color: '#98A1B2' }} weight={500}>
          Blockchain.com trading account
        </Text>
        <Text size='14px' lineHeight='21px' color='white' weight={500}>
          {walletAddress}
        </Text>
      </AccountInfo>
    </WalletWrapper>
  )
}

export default WalletItem
