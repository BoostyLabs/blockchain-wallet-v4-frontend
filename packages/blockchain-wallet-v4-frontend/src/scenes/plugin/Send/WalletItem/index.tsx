import React from 'react'
import styled from 'styled-components'

import { Icon, Text } from 'blockchain-info-components'
import CryptoAddress from 'components/CryptoAddress/CryptoAddress'

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  gap: 16px;
  border-radius: 8px;
  cursor: pointer;
  &:hover {
    background: ${(props) => props.theme.exchangeLogin};
  }
`

const Label = styled(Text)`
  margin: 16px 0;
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  color: ${(props) => props.theme.white};
`

type WalletItemProps = {
  address: string
  chooseAddress: (address: string) => void
}

const WalletItem: React.FC<WalletItemProps> = ({ address, chooseAddress }) => {
  return (
    <Wrapper onClick={() => chooseAddress(address)}>
      <Icon size='24px' name='ETH' />
      <Label>
        <CryptoAddress>{address}</CryptoAddress>
      </Label>
    </Wrapper>
  )
}

export default WalletItem
