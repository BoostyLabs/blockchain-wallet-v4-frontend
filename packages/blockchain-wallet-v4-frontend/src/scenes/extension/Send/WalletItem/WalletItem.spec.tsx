import React from 'react'
import WalletItem from 'blockchain-wallet-v4-frontend/src/scenes/extension/Send/WalletItem'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('WalletItem component', (): void => {
  it('Wallet item component renders correctly', (): void => {
    const walletAddress = '0x0d7dB08D75679bbE90b7B1eDfB8BE3a16897Ee17'
    const component = shallow(<WalletItem walletAddress={walletAddress} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
