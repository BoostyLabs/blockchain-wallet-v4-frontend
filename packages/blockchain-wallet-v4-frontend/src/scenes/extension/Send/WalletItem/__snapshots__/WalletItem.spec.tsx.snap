// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletItem component Wallet item component renders correctly 1`] = `
<styled.div
  id="walletItem"
>
  <SvgIconBlockchainCircle
    color="#473BCB"
    height="32px"
    width="32px"
  />
  <styled.div>
    <Text
      altFont={false}
      capitalize={false}
      color="grey700"
      cursor="inherit"
      italic={false}
      lineHeight="18px"
      opacity={1}
      size="12px"
      style={
        Object {
          "color": "#98A1B2",
        }
      }
      uppercase={false}
      weight={500}
    >
      Blockchain.com trading account
    </Text>
    <Text
      altFont={false}
      capitalize={false}
      color="white"
      cursor="inherit"
      italic={false}
      lineHeight="21px"
      opacity={1}
      size="14px"
      uppercase={false}
      weight={500}
    >
      0x0d7dB08D75679bbE90b7B1eDfB8BE3a16897Ee17
    </Text>
  </styled.div>
</styled.div>
`;
